File: RdFinanceBD/src/ca/mss/rd/mortgage/impl/MortgageSettings.java
package ca.mss.rd.mortgage.impl;

import java.math.MathContext;

import ca.mss.rd.mortgage.MortgageType;


public class MortgageSettings {
	final static public String className = MortgageSettings.class.getName();
	final static public long serialVersionUID = className.hashCode();
	final static public int SCALE = 2;
	final static public int SCALE_PRC = 2;
	final static public MathContext MATH_CONTEXT = new MathContext(SCALE);
	private MortgageType compoundType;

	/**
	 * Canadian mortgage by default
	 */
	public MortgageSettings() {
		this.compoundType = MortgageType.CANADIAN;
	}

	/**
	 * @return the compoundType
	 */
	public final MortgageType getCompoundType() {
		return compoundType;
	}

	/**
	 * @param compoundType the compoundType to set
	 */
	public final void setCompoundType(MortgageType compoundType) {
		this.compoundType = compoundType;
	}
}

File: RdFinanceBD/src.cpp/MortgageSettings.cpp
#include "MortgageSettings.h"

namespace ca
{
	namespace mss
	{
		namespace rd
		{
			namespace mortgage
			{
				namespace impl
				{
					using ca::mss::rd::mortgage::MortgageType;
const std::wstring MortgageSettings::className = MortgageSettings::typeid::getName();
java::math::MathContext *const MortgageSettings::MATH_CONTEXT = new java::math::MathContext(SCALE);

					MortgageSettings::MortgageSettings()
					{
						this->compoundType = MortgageType::CANADIAN;
					}

					MortgageType *MortgageSettings::getCompoundType()
					{
						return compoundType;
					}

					void MortgageSettings::setCompoundType(MortgageType *compoundType)
					{
						this->compoundType = compoundType;
					}
				}
			}
		}
	}
}
